### PR TITLE
Add setup privileged project util

### PR DESCRIPTION
**Pull request by Phaow**
## Add setup privileged project util
PodSecurity restricted enabled by default on 4.12, used hostpath cases need namespace set to privilieged. Add setup privileged project util for using privilieged namespace scenarios.

**[Flake record](https://${report-portal-Domain}/ui/#prow/launches/454/207510/12395570/12395797/log?item1Params=filter.eq.hasStats%3Dtrue%26filter.eq.hasChildren%3Dfalse%26filter.in.type%3DSTEP%26filter.in.status%3DFAILED%252CINTERRUPTED)**
**[Test record ](https://${jenkins-Domain}/job/ocp-common/job/Runner/577369/console)**

### CODE REVIEW

**Comment by Phaow** [comment]
@duanwei33 @chao007 @ropatil010 @radeore Could you please help to have a review when you get a chance? Thanks!

File: features/step_definitions/project.rb [hunk 1]
```diff
@@ -34,6 +34,47 @@
...
+      step 'I create a new project'
+      unless @result[:success]
+        logger.error(@result[:response])
+        raise "unable to create project, see log"
+      end
+    else
+      cache_resources *projects
+    end
+  end
+  if env.version_ge("4.12", user: user)
```
Comment (duanwei33) [inline features/step_definitions/project.rb:65]: @Phaow  If we only add this for 4.12+, the step name makes confused for the previous release, right?

Comment (Phaow) [reply features/step_definitions/project.rb:65]: @duanwei33 Yeah. TBH, I don't have a better step name for this. How about `Given /^I have a project using hostpath`?

Comment (duanwei33) [reply features/step_definitions/project.rb:65]: @Phaow  I agree it is not easy to backport.
From the code side, I prefer to apply them on all releases (if this cmd works), and let the user choose to use it or not in different releases. So the code and the step name are consistent.
Like in 4.11-, users could use "I have a project" if they don't need a privileged one.

Comment (duanwei33) [reply features/step_definitions/project.rb:65]: Adding @liangxia may give us some suggestions here.

Comment (Phaow) [reply features/step_definitions/project.rb:65]: Sure. Thanks!

Comment (duanwei33) [reply features/step_definitions/project.rb:65]: @Phaow Another way(I think it is better one) is we could add an individual step like "labeling () project as pod-security privileged", then we could use this step after "I have a project".

Comment (Phaow) [reply features/step_definitions/project.rb:65]: Good option! I'll have a try.

Comment (liangxia) [reply features/step_definitions/project.rb:65]: How about `Given /^I have a project with proper privilege$/` ?
```
Given /^I have a project with proper privilege$/ do
  step %Q/I have a project/
  if env.version_ge("4.12", user: user)
    blabla
  end
end

Comment (Phaow) [reply features/step_definitions/project.rb:65]: Perfect. Many thanks!

**Comment by Phaow** [comment]
Hi @duanwei33 , updated and retest passed, could you please help take a look when you get a chance? Thanks!
**[Test record](https://$(jenkins_domain)/job/ocp-common/job/Runner/578729/console)**

**Comment by Phaow** [comment]
Seems the build image failed not related the changed code.
failed at step `$ bundle install --gemfile=/verification-tests/tools/Gemfile`
```
Bundler could not find compatible versions for gem "net-ssh":
  In Gemfile:
    net-ssh

    net-scp was resolved to 3.0.0, which depends on
      net-ssh (< 7.0.0, >= 2.6.5)
error: build error: error building at STEP "RUN set -x &&     mv /tierN/ /verification-tests/features/tierN/ &&     chgrp -R 0 /verification-tests &&     chmod -R g=u /verification-tests &&     /verification-tests/tools/install_os_deps.sh &&     /verification-tests/tools/hack_bundle.rb &&     yum clean all -y &&     rm -rf /var/cache/yum /var/tmp/* /tmp/* /verification-tests/Gemfile.lock": error while running runtime: exit status 1
```

**Pull request merged**